integration/spark-interfaces-scala/src/main/scala/io/openlineage/spark/builtin/planNodes.scala

```diff
@@ -0,0 +1,111 @@
+/** Copyright 2018-2023 contributors to the OpenLineage project
+  * SPDX-License-Identifier: Apache-2.0
+  */
+package io.openlineage.spark.builtin.nodes
+
+import io.openlineage.client.OpenLineage.DatasetFacetsBuilder
+import io.openlineage.client.OpenLineage.InputDatasetInputFacetsBuilder
+import io.openlineage.client.OpenLineage.OutputDatasetOutputFacetsBuilder
+import io.openlineage.client.utils.DatasetIdentifier;
+import io.openlineage.spark.builtin.common.OpenLineageContext;
+
+/** Trait to be implemented for LogicalPlan nodes to extract lineage information
+  * about input datasets.
+  */
+trait InputLineageNode {
```

Comment:
I was thinking at the best way to evolve those API. It will probably be to create a new trait every time we make a change to avoid issues.
To that point it could be a good idea to put it in an "experimental" package so that we can create the final version without compatibility issues when we are settled on the API.

Reply:
I moved all the traits into `io.openlineage.spark.builtin.scala.v1` so that we can publish Java version of it if someone wants it, and we can release v2 in case of breaking changes required. I didn't want to include `experimental` within the package name, as this can discourage potential users.